tools/ice_env.sh

```diff
@@ -0,0 +1,143 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+CONTAINER_NAME="ice_env"
+
+setup_docker_image() {
+    echo "Europe/Berlin" > /etc/timezone
+    ln -sf /usr/share/zoneinfo/Europe/Berlin /etc/localtime
+
+    local BASE_OS_VERSION=$(echo $OS_VERSION | sed -n "s/\([^\:]*\).*/\1/p")
+
+    if [[ $BASE_OS_VERSION == "ubuntu" ]]; then
+        apt update
+        apt -y install libbison-dev g++ gcc sudo cmake git fish gdb lldb llvm clang clang-format
+    elif [[ $BASE_OS_VERSION == "archlinux" ]]; then
+        pacman -Syu --noconfirm base base-devel clang cmake git fish gdb lldb llvm
+    else 
+        echo Please install the following packages to have a working iceoryx environment
+        echo libbison-dev g++ gcc sudo cmake git fish gdb lldb llvm clang clang-format
+    fi
+
+    mkdir -p /root/.config/fish
+    echo "set -gx PATH /iceoryx/tools/ci /iceoryx/scripts \$PATH" >> /root/.config/fish/config.fish
+    echo "set -gx ASAN_OPTIONS 'symbolize=1,detect_leaks=1,abort_on_error=1,quarantine_size_mb=8192'" >> /root/.config/fish/config.fish
+    echo "set -gx UBSAN_OPTIONS 'print_stacktrace=1'" >> /root/.config/fish/config.fish
+    echo "set -gx ASAN_SYMBOLIZER_PATH '/usr/bin/llvm-symbolizer'" >> /root/.config/fish/config.fish
+
+    exit
+}
+
+start_docker_session() {
+    bash
+    exit
+}
+
+help() {
+    echo
+    echo "iceoryx development environment help"
+    echo
+    echo "  $0 [ACTION] (optional)[OS_VERSION]"
+    echo
+    echo "ACTION:"
+    echo "  start          - start iceoryx development environment"
+    echo "  stop           - stops the iceoryx development environment"
+    echo "  enter          - enters the iceoryx development environment, if it is"
+    echo "                   not running it will be started first"
+    echo
+    echo "OS_VERSION:"
+    echo "  A string which will be forwarded to \"-t\" in the docker command."
+    echo "  The version of operating system to load. Default value is ubuntu:20.04."
+    echo "  Other possibilities (not all) are:"
+    echo "    ubuntu:18.04"
+    echo "    archlinux"
+    echo
+    echo "Example:"
+    echo "  $0 start archlinux     # starts iceoryx environment with archlinux docker container"
+    echo "  $0 enter ubuntu:18.04  # enters (and starts if not running) iceoryx environment with ubuntu"
+    echo
+    exit
+}
+
+start_docker() {
+    local ICEORYX_PATH=$(git rev-parse --show-toplevel)
+    if [[ $(docker container inspect -f '{{.State.Running}}' $CONTAINER_NAME 2>/dev/null) == "true" ]]; then
+        echo iceoryx development environment already running \(docker container: $CONTAINER_NAME\)
+        exit
+    fi
+
+    docker run --name $CONTAINER_NAME \
+               --mount type=bind,source=${ICEORYX_PATH},target=/iceoryx \
+               --hostname ${OS_VERSION} \
+               -dt --memory "4g" ${OS_VERSION}
```

Comment:
It would be good to set the possible max. shared memory size with `--shm-size 2g` to a certain value. Default is 64 Mb which is not enough.